### DISCUSSION TITLE
MQTT availability topic and remove 'retained' option

### DISCUSSION THREAD

**Discussion by VelocityMaximus** [serkri/SmartEVSE-3#189] [Ideas]
It would be nice if there is an availability topic for MQTT that states the EVSE is online.

With the current implementation you can not determine if the EVSE is online or not. With an availability topic there is a watchdog function in MQTT that if there has been no update after a certain time that the topic value goes from online to offline automatic, stating that the data is not availble. This feature also shows in Home Assistant that the EVSE went offline on MQTT and will be 'unavailable' 

Also retained MQTT messages can give a wrong indication about the current state because they are always available, even if the EVSE is offline for a longer time. This is due to stored values in the MQTT broker. (maybe some topics can be handy to have them retained like the 'Mode' and 'MaxCurrent' if they are controlled via automations in HA)

**Comment by dingo35** [comment]
What topics are currently retained that you think should not?

**Comment by VelocityMaximus** [comment]
The topics that could be retained are those that are valuable when there are issues like wifi disconnects for unknown reasons.
If possible, make an option to set retained as an option on state topics via the webinterface.

The topic connected can be used for the watchdog.
The default state for this topic should be then offline and set by a loop timer ie. every 10 seconds to online.
When the topic is not updated, the mqtt protocol reverts to the offline value.
With HA you can then use this topic to each topic that is subscribed and show it as unavailable instead of a non updated state.

With the current setup it is not clear if the ESVE is connected to MQTT. In practice it can happen that it looks like the EV is charging but has stopped due to an error and went offline.

With the latest version of the ESKRI firmware, the state topic connected goes from offline to online but is never updated and is not usable as availability topic.

**Comment by dingo35** [comment]
That is a very general story but no specific answer to my question. 
Making retainability configurable IMHO makes no sense, it complicates software and user interface and only avoids making good design decisions.

If the availability topic is not to your liking, contact the author of the PR and/or write your own....
